pkg/controllers/state/suite_test.go

```diff
@@ -1422,6 +1423,141 @@ var _ = Describe("Consolidated State", func() {
 	})
 })
 
+var _ = Describe("Taints", func() {
+	var nodeClaim *v1beta1.NodeClaim
+	var node *v1.Node
+	BeforeEach(func() {
+		instanceType := cloudProvider.InstanceTypes[0]
+		nodeClaim, node = test.NodeClaimAndNode(v1beta1.NodeClaim{
+			ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{
+				v1.LabelInstanceTypeStable: instanceType.Name,
+			}},
+			Status: v1beta1.NodeClaimStatus{
+				ProviderID: test.RandomProviderID(),
+			},
+		})
+	})
+	Context("Managed", func() {
+		It("should not consider ephemeral taints on a managed node that isn't initialized", func() {
+			node.Spec.Taints = []v1.Taint{
+				{Key: v1.TaintNodeNotReady, Effect: v1.TaintEffectNoSchedule},
+				{Key: v1.TaintNodeUnreachable, Effect: v1.TaintEffectNoSchedule},
+				{Key: cloudproviderapi.TaintExternalCloudProvider, Effect: v1.TaintEffectNoSchedule, Value: "true"},
+			}
```

Comment:
Worth using the same ephemeral taints list variable here so that this is automatically updated when we change that variable?

Reply:
I don't love using constants that are defined within your own package in testing since it can mask bugs. Imagine that we accidentally change the ephemeralTaints value in the state package and now it doesn't include one of these taints. That wouldn't cause a testing failure when I'd argue that it should since you are now not considering one of the taints that you had statically intended to consider.

Reply:
Discussed offline: we both think that it's reasonable to use the static value in testing to avoid cases like this.